src/super_gradients/training/samples/depth_estimation_sample.py

```diff
@@ -0,0 +1,28 @@
+import dataclasses
+from typing import Union
+
+import numpy as np
+import torch
+from PIL.Image import Image
+
+__all__ = ["DepthEstimationSample"]
+
+
+@dataclasses.dataclass
+class DepthEstimationSample:
+    """
+    A dataclass representing a single depth estimation sample.
+    Contains input image and depth map.
+
+    :param image:              Image of [H, W, (C if colorful)] shape.
+    :param depth_map:          Depth map of [H, W] shape.
+    """
+
+    __slots__ = ["image", "depth_map"]
+
+    image: Union[np.ndarray, torch.Tensor]
+    depth_map: Union[np.ndarray, torch.Tensor]
+
+    def __init__(self, image: Union[np.ndarray, torch.Tensor, Image], depth_map: Union[np.ndarray, torch.Tensor, Image]):
```

Comment:
I think we can show some love to the user and should check whether depth_map is indeed a [H,W] shape and there is no dummy channel dimension (And squeeze if it is there). This way we can ensure consistency between the documentation and the state of the sample.
Of course if the channel dimension > 1 that we should throw an error.